src/vivarium/framework/components/manager.py

```diff
@@ -254,7 +264,10 @@ def apply_configuration_defaults(self, component: Component | Manager) -> None:
             )
         except DuplicatedConfigurationError as e:
             new_name, new_file = component.name, self._get_file(component)
-            old_name, old_file = e.source, self._get_file(self.get_component(e.source))
+            if e.source:
+                old_name, old_file = e.source, self._get_file(self.get_component(e.source))
+            else:
+                old_name, old_file = "another component", "another file"
```

Comment:
clever. It does lead to a slightly awkward error message "...but it has already been set by another component in file another file".

Not sure if the word "file" should be removed in the message or not below.